integration/query_comparision_test.go

```diff
@@ -986,8 +986,84 @@ func TestQueryComparisonLte(t *testing.T) {
 	}
 }
 
+func TestQueryComparisonNin(t *testing.T) {
+	t.Parallel()
+	providers := []shareddata.Provider{shareddata.Scalars, shareddata.Composites}
+	ctx, collection := setup(t, providers...)
+
+	var scalarDataTypesFilter bson.A
+	for _, scalarDataType := range shareddata.Scalars.Docs() {
+		scalarDataTypesFilter = append(scalarDataTypesFilter, scalarDataType.Map()["value"])
+	}
+
+	var compositeDataTypesFilter bson.A
+	for _, compositeDataType := range shareddata.Composites.Docs() {
+		compositeDataTypesFilter = append(compositeDataTypesFilter, compositeDataType.Map()["value"])
+	}
+
+	for name, tc := range map[string]struct {
+		q           bson.D
```

Comment:
could you please rename `q` to `filter`?

Reply:
Better yet, remove the common part: `filter := bson.D{{"value", bson.D{{"$nin", tc.value}}}}`. In that case it should be `value`, not `filter` (not the full value of the second argument) or `q` (which we use more generically)

Reply:
will do

Reply:
fixed